internal/pkg/otel/components.go

```diff
@@ -75,7 +70,7 @@ func components(extensionFactories ...extension.Factory) func() (otelcol.Factori
 		factories := otelcol.Factories{}
 
 		// Receivers
-		factories.Receivers, err = receiver.MakeFactoryMap(
```

Comment:
[**For reviewers**] (CC: @swiatekm) The `receiver#MakeFactoryMap` seems to have been deprecated in v0.120.0 in favor of `otelcol#MakeFactoryMap` [causing lint for this PR to fail](https://github.com/elastic/elastic-agent/actions/runs/13654554626/job/38170471376?pr=7172). However, I don't understand why the lint for [version up PR](https://github.com/elastic/elastic-agent/pull/7050) to `v0.120.0` [succeeded](https://github.com/elastic/elastic-agent/actions/runs/13632614286/job/38103618099).

Reply:
The linter only runs on PR changes. I didn't change this file in my update, while you have.

Reply:
Ah, I suspected as much. Thanks for confirming.